tests/api_test.py

```diff
@@ -6377,6 +6377,38 @@ def transposed(y):
   return transposed
 
 
+# This class wraps api.custom_transpose in order to pass in a
```

Comment:
Maybe not in this PR, but given that api_test.py is gigantic, I'm starting to think we should factor out CustomTransposeTest (and perhaps CustomJVPTest / CustomVJPTest).